.github/workflows/scheduled.yml

```diff
@@ -63,8 +70,54 @@ jobs:
           cask: false
           test-bot: false
 
-      - name: Check formula for missing upstreams etc.
-        run: brew audit --online --skip-style --only github_repository_archived,gitlab_repository_archived,homepage ${{ matrix.formula }}
+      - name: Check formula source is not archived.
+        id: archived
+        if: matrix.os != 'macos-latest'
+        run: brew audit --online --skip-style --only github_repository_archived,gitlab_repository_archived ${{ matrix.formula }}
+
+      - name: Report online issues
+        if: ${{ failure() && steps.archived.conclusion == 'failure' }}
+        run: |
+          gh issue comment "$REPORTING_ISSUE" \
+                           --repo homebrew/homebrew-core \
+                           --body "${{ matrix.formula }} should be archived. Check ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+
+      - name: Check formula for unavailable homepage.
+        id: homepage
+        if: matrix.os != 'macos-latest'
+        run: brew audit --online --skip-style --only homepage ${{ matrix.formula }}
+
+      - name: Report homepage issues
+        if: ${{ failure() && steps.homepage.conclusion == 'failure' }}
+        run:  |
+          gh issue comment "$REPORTING_ISSUE" \
+                           --repo homebrew/homebrew-core \
+                           --body "${{ matrix.formula }} has homepage issues. Check ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
 
       - name: Check formula for missing sources.
+        id: fetch
+        if: matrix.os != 'macos-latest'
         run: brew fetch -s ${{ matrix.formula }}
+
+      - name: Report fetch issues
+        if: ${{ failure() && steps.fetch.conclusion == 'failure' }}
+        run:  |
+          gh issue comment "$REPORTING_ISSUE" \
+                           --repo homebrew/homebrew-core \
+                           --body "${{ matrix.formula }} source has problems. Check ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+
+      - name: Try and build formula.
+        id: build
+        env:
+          HOMEBREW_NO_INSTALL_FROM_API: '1'
+        run: brew install -s ${{ matrix.formula }}
+
+      # TODO: Re-enable once it skips formulae that are not fir the current OS
+      # - name: Report build issues
+      #   if: ${{ failure() && steps.build.conclusion == 'failure' }}
+      #   run:  |
+      #     gh issue comment "$REPORTING_ISSUE" \
+      #                      --repo homebrew/homebrew-core \
+      #                      --body "${{ matrix.formula }} failed to build on ${{ matrix.os }}. Check ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+
+
```

Comment:
Sorry, missed this before: I don't think we should do this. It's going to use _way_ more CI than the other approaches, some formulae will literally always time out and we find this stuff out when we do regular yearly rebottling.

Reply:
I'll open this as a separate PR for this, but I do believe that it will save us a bunch of effort in those times where we rebottle things.

Reply:
@SMillerDev I think we'll need to exclude the `long-build` formulae, handle macOS requirements properly here, use our macOS runners but only when not being already used (as some stuff will fail on GitHub Actions macOS runners and not ours and vice-versa). I think it would be desirable but just setting expectations that I think the level of work required for it to not be annoying and/or a waste of resources is fairly high.